### ISSUE TITLE
Feature importances Nan

### ISSUE THREAD

**Issue by gsndr** [ModelOriented/DALEX#415]
When I use model_parts on my model (a Keras Neural Network) in mosto case the results of the dropout_loss is Nan.
can you explain to me what this results depends on?

**Comment by hbaniecki** [comment]
Could you provide a minimal code example (with sample data) to reproduce this problem?

**Comment by gsndr** [comment]
The training set is very big more than 1.000.000 of data. If I pass a reduced dataset as input to the creation of the Explainer (dx.Explainer()) .
My question are: 
1) Why the  framework returns this Nan result when the data are big?
2) Could be the results of the explainer biased by passing this reduced dataset?

**Comment by hbaniecki** [comment]
1. I see no causation between the size of the data and the `NaN` results. `dropout_loss` is computed as a difference between the loss value on a baseline dataset and the loss value on a dataset with a permuted variable (column), see the algorithm details in [EMA book](https://ema.drwhy.ai/featureImportance.html). As the loss for classification models is `1-auc`, it might be that the `auc` returned `NaN` due to wrong model output / `y`. `model_parts` has a `loss_function` parameter which may be a custom function to compute a measure for the `dropout_loss`.
2. In `model_parts`, `predict_parts` and `model_profile` explanations there is a `N` parameter that denotes the size of the dataset used for explanation computation. In the case of `model_profile` it is usually set to `500-1000`; in the case of `model_parts` it might be enough to sample `1000-3000` observations, which depends on the size of data and variable distributions. Answering the question: explanations may be biased when calculated on reduced datasets (depending on multiple factors) but it is a standard practice to speed up the computation. One can compute explanations on multiple bootstrap samples to achieve statistically significant results.

I would be able to debug the code, provided the example.

**Comment by gsndr** [comment]
Hi,
I tried to reproduce the error:
 `import dalex as dx
import numpy as np
from keras.models import load_model

X= np.load('X.npy')
Y = np.load('Y.npy')
print(Y.shape)
model=load_model('model.h5')

explainer = dx.Explainer(model, X, Y,model_type='classification')

seed=123
feature = explainer.model_parts(random_state=seed)

r = feature.result
print(r)`

The model and the subset of the dataset is shared here : https://drive.google.com/drive/folders/1lOLPs2e8Ff65IzvEZPSRiKaGbUhItRM-?usp=sharing

The dropout_loss is Nan for each feature

**Comment by hbaniecki** [comment]
Hi @gsndr, thanks for pointing this issue. As seen in a warning message: `dx.Explainer` did not properly detect the model type, because we were considering only `tensorflow.keras` models and not older `keras` itself. This should be fixed now.
 
![image](https://user-images.githubusercontent.com/32574004/120233888-0c587880-c257-11eb-8186-4946a81c09c9.png)

![image](https://user-images.githubusercontent.com/32574004/120239114-e387b180-c25d-11eb-9b0a-2ef8463047ba.png)

**Comment by hbaniecki** [comment]
`dalex` v1.2.0 is now available on PyPI `pip install dalex -U`
